core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/validation/ValidateRequest.java

```diff
@@ -137,7 +140,11 @@ private ValidationResult doValidate(BranchContext context, Writer index) throws
 								.execute(context)
 								.getItems();
 						
+						
 						final Set<ComponentIdentifier> existingComponentIdentifiers = existingRuleIssues.stream().map(ValidationIssue::getAffectedComponent).collect(Collectors.toSet());
+						
+						final Map<ComponentIdentifier, ValidationIssue> existingIsssuesByComponentIdentifier = existingRuleIssues.stream().collect(Collectors.toMap(ValidationIssue::getAffectedComponent, Function.identity()));
```

Comment:
If you create an `affectedComponent` to `issue` `Map`, then you can safely remove the `existingComponentIdentifiers` variable and use the `keySet` of the `Map`.